doc/gazebo_simulation/gazebo_simulation.rst

```diff
@@ -0,0 +1,351 @@
+
+4. Add friction and colorize the links
+--------------------------------------
+In order to have a nice illustration of the robot in Gazebo simulation we need to colorize the links. 
+Moreover friction forces are added in order to have realistic dynamics. You can ignore them at all or change their values to experiment.
+Since the focus is MoveIt in this tutorial, we will just use the values from the provided solution.
+
+This step is a bit tedious to do manually, so the ultimate :code:`xacro` file is provided entirely in below:
+
+.. code-block:: xml
+
+    <!-- panda.gazebo.xacro -->
+    <?xml version='1.0' encoding='utf-8'?>
+    <robot xmlns:xacro="http://www.ros.org/wiki/xacro">
+        <xacro:macro name="panda_gazebo" params="arm_id">
+            <xacro:macro name="arm_gazebo" params="link">
+                <gazebo reference="${link}">
+                    <material>Gazebo/White</material>
+                    <mu1>0.2</mu1>
+                    <mu2>0.2</mu2>
+                </gazebo>
+            </xacro:macro>
+            <xacro:macro name="hand_gazebo" params="link">
+                <gazebo reference="${link}">
+                    <material>Gazebo/Grey</material>
+                    <mu1>0.2</mu1>
+                    <mu2>0.2</mu2>
+                </gazebo>
+            </xacro:macro>
+            <xacro:arm_gazebo link="${arm_id}_link0"/>
+            <xacro:arm_gazebo link="${arm_id}_link1"/>
+            <xacro:arm_gazebo link="${arm_id}_link2"/>
+            <xacro:arm_gazebo link="${arm_id}_link3"/>
+            <xacro:arm_gazebo link="${arm_id}_link4"/>
+            <xacro:arm_gazebo link="${arm_id}_link5"/>
+            <xacro:arm_gazebo link="${arm_id}_link6"/>
+            <xacro:hand_gazebo link="${arm_id}_link7"/>
+            <xacro:hand_gazebo link="${arm_id}_link8"/>
+            <xacro:hand_gazebo link="${arm_id}_hand"/>
+            <xacro:hand_gazebo link="${arm_id}_rightfinger"/>
+            <xacro:hand_gazebo link="${arm_id}_leftfinger"/>
+        </xacro:macro>
+    </robot>
+
+Then add the following block to the end of :code:`franka_description/robots/panda_arm_hand_urdf.xacro` file:
+
+.. code-block:: xml
+
+    <xacro:include filename="$(find franka_description)/robots/panda.gazebo.xacro"/>
+    <xacro:panda_gazebo arm_id="panda"/>
+
+5. Configure gazebo_ros_control, transmissions and actuators
+------------------------------------------------------------
+
+This is necessary for the robot to move in Gazebo. ROS Control is a highly capable robot-agnostic stack, providing interfaces
+to control theoretically any type of robot. :code:`gazebo_ros_control` enables the ROS control to be used in Gazebo. 
+See `its document <http://gazebosim.org/tutorials/?tut=ros_control>`_ for full details.
+
+Add the following to the newly created :code:`panda.gazebo.xacro` file:
```

Comment:
Where in the file? Inside the `<robot>` tag?

Reply:
Just before `</robot>`. I assumed some level of intimacy..

Reply:
It's a tutorial that many first-time users will read though, and it's easy to trip up and get lost on "that should be obvious" things. Better to err on the side of too much detail. Don't forget that if you can write a tutorial in the first place, you automatically tend to underestimate the difficulty.